generic_tracers/generic_COBALT.F90

```diff
@@ -870,150 +861,347 @@ subroutine user_add_params(param_file)
+    ! Note: Values for these parameters are entered in day-1 (micromol kg-1)-1 and converted to sec-1 (mol kg-1)-1
     !
-    call get_param(param_file, "generic_COBALT", "mort_Sm",phyto(SMALL)%mort,  "mort_Sm",units="day-1", &
-                   default= 0.0, scale = I_sperd ) ! s-1
-    call get_param(param_file, "generic_COBALT", "mort_Di",phyto(DIAZO)%mort,  "mort_Di",units="day-1", &
-                   default= 0.0, scale = I_sperd ) ! s-1
-    call get_param(param_file, "generic_COBALT", "mort_Lg",phyto(LARGE)%mort,  "mort_Lg",units="day-1", &
-                   default= 0.0, scale = I_sperd ) ! s-1
-    call get_param(param_file, "generic_COBALT", "mort_Md",phyto(MEDIUM)%mort, "mort_Md",units="day-1", &
-                   default= 0.0, scale = I_sperd ) ! s-1
+    ! References:
+    ! Jackson, 1990 (https://doi.org/10.1016/0198-0149(90)90038-W)
+    ! Waite et al., 1992 (https://doi.org/10.1007/BF00350862)
+    !
+    call get_param(param_file, "generic_COBALT", "agg_Sm", phyto(SMALL)%agg, &
+                   "aggregation rate constant for small phytoplankton", units="day-1 (micromol N kg)-1", &
+                   default=0.05, scale = micromol2mol/sperd)
+    ! Diazotrophs, which are assumed not to aggregate, are modeled after trichodesmium
+    call get_param(param_file, "generic_COBALT", "agg_Di", phyto(DIAZO)%agg, &
+                   "aggregation rate constant for diazotrophs", units="day-1 (micromol N kg)-1", & 
+                   default=0.0 , scale = micromol2mol/sperd)
+    call get_param(param_file, "generic_COBALT", "agg_Lg", phyto(LARGE)%agg, &
+                   "aggregation rate constant for large phytoplankton", units="day-1 (micromol N kg)-1", &
+                   default=0.25, scale = micromol2mol/sperd)
+    call get_param(param_file, "generic_COBALT", "agg_Md", phyto(MEDIUM)%agg, &
+                   "aggregation rate constant for medium phytoplankton", units="day-1 (micromol N kg)-1", &
+                   default=0.10, scale = micromol2mol/sperd)
+    !
+    ! Phytoplankton viral losses are assumed to be a density-dependent (quadratic) loss term that is more effective
+    ! for small phytoplankton (Murray and Jackson, 1992).  Rates are temperature dependent and assumed to impact both
+    ! phytoplankton and free-living heterotrophic bacteria. For bacteria, rates were set such that ~20% of bacterial
+    ! production is lost to viral lysis (e.g., Fuhrman, 2000; Suttle, 2005).  Viral mortality for small phytoplankton
+    ! (i.e., picophytoplankton) was assumed similar to bacteria, with rates for larger phytoplankton scaled downward
+    ! in accordance with expected decreasing efficacy as surface area to volume ratios decrease.  Viruses are assumed
+    ! active regardless of the stress levels of the cell.
+    ! Note: Values for these parameters are entered in day-1 (micromol kg-1)-1 and converted to sec-1 (mol kg-1)-1
```

Comment:
@charliestock again here the unit is not consistent with what we put later in `get_param` call (https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/ocean_BGC/blob/2aa27adc80da33051cce0dc81f6b688db4d91cd0/generic_tracers/generic_COBALT.F90#L964)

Reply:
Hi YC,  Thanks for catching this.  They should indeed be "day-1 (micromol N kg-1)-1" in both cases.  I will put in a new commit to address this.  The rationale for the way I wrote this is that the units are effectively "per day, per unit concentration".  The grouping of the units is intended to convey this.